src/seam-connect/client.ts

```diff
@@ -52,33 +66,67 @@ export class Seam extends Routes {
   constructor(apiKeyOrOptions?: string | SeamClientOptions) {
     super()
 
-    const { apiKey, endpoint, workspaceId, axiosOptions } =
-      getSeamClientOptionsWithDefaults(apiKeyOrOptions)
-
-    const isRegularAPIKey = apiKey?.startsWith("seam_")
-
-    if (isRegularAPIKey && workspaceId)
-      throw new Error(
-        "You can't use API Key Authentication AND specify a workspace. Your API Key only works for the workspace it was created in. To use Session Key Authentication with multi-workspace support, contact Seam support."
-      )
-
-    if (!apiKey) {
-      throw new Error(
-        "SEAM_API_KEY not found in environment, and apiKey not provided"
+    const {
+      apiKey,
+      endpoint,
+      workspaceId,
+      axiosOptions,
+      clientAccessToken,
+      clientSessionToken,
+    } = getSeamClientOptionsWithDefaults(apiKeyOrOptions)
+    let bearer = `Bearer `
```

Comment:
This large nested if/else block is hard to parse. Perhaps easier to maintain if we move this to a function, then we can do early return. 🐧 E.g.,

```
const getBearerToken({ clientSessionToken, apiKey }) => {
   if (apiKey || clientAccessToken) {
     throw new Error(
        "You can't use clientSessionToken AND specify apiKey or clientAccessToken."
      )
    }
    if (clientSessionToken} {
      if (!clientSessionToken?.startsWith("seam_cst")) throw
      return clientSessionToken
    }
```

Reply:
agree that this function is a bit hard to parse and early return would work better (especially because this `bearer` var is being mutated)